src/Control/Concurrent/Async/Lifted/Safe.hs

```diff
@@ -306,25 +306,28 @@ waitBoth = Unsafe.waitBoth
 race
   :: forall m a b. (MonadBaseControl IO m, Forall (Pure m))
   => m a -> m b -> m (Either a b)
-race = Unsafe.race
-  \\ (inst :: Forall (Pure m) :- Pure m a)
-  \\ (inst :: Forall (Pure m) :- Pure m b)
+race = unliftFuncs A.race
 
 -- | Generalized version of 'A.race_'.
 race_
   :: forall m a b. (MonadBaseControl IO m, Forall (Pure m))
   => m a -> m b -> m ()
-race_ = Unsafe.race_
-  \\ (inst :: Forall (Pure m) :- Pure m a)
-  \\ (inst :: Forall (Pure m) :- Pure m b)
+race_ = unliftFuncs A.race_
 
 -- | Generalized version of 'A.concurrently'.
 concurrently
   :: forall m a b. (MonadBaseControl IO m, Forall (Pure m))
   => m a -> m b -> m (a, b)
-concurrently = Unsafe.concurrently
-  \\ (inst :: Forall (Pure m) :- Pure m a)
-  \\ (inst :: Forall (Pure m) :- Pure m b)
+concurrently = unliftFuncs A.concurrently
+
+-- | Helper function which converts an 'IO'-specialized combinator
+-- into a generalized combinator.
+unliftFuncs :: forall m a b c. (MonadBaseControl IO m, Forall (Pure m))
```

Comment:
Could we rename `unliftFuncs` to `liftBaseOp2_`? It's more like `liftBaseOp_` in `monad-control` but the argument function takes two params and also it has `Forall (Pure m)` constraint.